api/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -62,6 +62,17 @@ func (r *OpenTelemetryCollector) ValidateCreate() error {
 	if r.Spec.Mode != ModeStatefulSet && len(r.Spec.VolumeClaimTemplates) > 0 {
 		return errors.New("Can only specify VolumeClaimTemplates for statefulsets")
 	}
+
+	// validate replicas for sidecar and daemonsets mode
+	if (r.Spec.Mode == ModeSidecar || r.Spec.Mode == ModeDaemonSet) && *r.Spec.Replicas > 0 {
+		return errors.New("Can only specify Replicas for deployments and statefulsets")
```

Comment:
It might be better to align the message with the check, like: `the OpenTelemetry Collector mode is set to %s, which does not support the attribute "replicas"`